inspire_disambiguation/core/ml/models.py

```diff
@@ -319,68 +585,118 @@ def __init__(self, estimator):
-        self.clustering_method = 'average'
+        self.clustering_method = "average"
+
+    def load_data(self, signatures, input_clusters):
+        """Loads data to the estimator vectors
 
-    def load_data(self, signatures_path, publications_path, input_clusters_path):
-        signatures_by_uuid = load_signatures(signatures_path, publications_path)
+        Args:
+            signatures (iterable): Signatures which should be processed
+            input_clusters (iterable): TODO
+
+        """
+        signatures_by_uuid = load_signatures(signatures)
 
         self.X = np.empty((len(signatures_by_uuid), 1), dtype=np.object)
         self.y = -np.ones(len(self.X), dtype=np.int)
 
         i = 0
-        with open(input_clusters_path, 'r') as fd:
-            for line in fd:
-                cluster = json.loads(line)
-                for signature_uuid in cluster['signature_uuids']:
-                    if signature_uuid not in signatures_by_uuid:
-                        continue  # TODO figure out how this can happen
-                    self.X[i, 0] = signatures_by_uuid[signature_uuid]
-                    self.y[i] = cluster['cluster_id']
-                    i += 1
+        for cluster in input_clusters:
+            for signature_uuid in cluster["signature_uuids"]:
+                if signature_uuid not in signatures_by_uuid:
+                    continue  # TODO figure out how this can happen
+                self.X[i, 0] = signatures_by_uuid[signature_uuid]
+                self.y[i] = cluster["cluster_id"]
+                i += 1
 
     def load_model(self, input_filename):
-        with open(input_filename, 'rb') as fd:
+        """Loads model dumped by pickle
+
+        Args:
+            input_filename (str): path to file with dumped ethnicity model.
+        """
+        with open(input_filename, "rb") as fd:
             self.clusterer = pickle.load(fd)
 
     def save_model(self, output_filename):
-        with open_file_in_folder(output_filename, 'wb') as fd:
+        """Dump object to a file
+
+        Args:
+            output_filename (str): Path where model will be dumped
+        """
+        with open_file_in_folder(output_filename, "wb") as fd:
             pickle.dump(self.clusterer, fd, protocol=pickle.HIGHEST_PROTOCOL)
 
     def fit(self, n_jobs=8):
+        """Fit data using the estimator"""
         self.clusterer = BlockClustering(
             blocking=self.block_function,
             base_estimator=ScipyHierarchicalClustering(
                 affinity=_affinity,
                 threshold=self.clustering_threshold,
                 method=self.clustering_method,
-                supervised_scoring=b3_f_score),
+                supervised_scoring=b3_f_score,
+            ),
             n_jobs=n_jobs,
-            verbose=True)
+            verbose=True,
+        )
         self.clusterer.fit(self.X, self.y)
 
 
 @attr.s(slots=True)
 class Signature(object):
+    """Signature model for estimators"""
```

Comment:
I wouldn't call it a model (at least it's not a ML model), it's a data structure, maybe even move it out to a separate file?